geoh5py/groups/property_group.py

```diff
@@ -346,7 +321,29 @@ def uid(self, uid: str | UUID):
 
         self._uid = uid
 
-    def verify_data(self, data: Data | UUID | str) -> UUID:
+    @staticmethod
+    def validate_association(
+        value: str | DataAssociationEnum | None, properties: list[Data] | None
+    ) -> DataAssociationEnum:
+        if properties is None and value is None:
+            raise ValueError(
+                "At least one of 'properties' or 'association' must be provided."
+            )
+
+        if value is None and properties is not None:
+            value = properties[0].association
+
+        if isinstance(value, str):
+            value = getattr(DataAssociationEnum, value.upper())
+
+        if not isinstance(value, DataAssociationEnum):
+            raise TypeError(
+                f"Association must be 'VERTEX', 'CELL' or class of type {DataAssociationEnum}"
+            )
+
+        return value
+
+    def validate_data(self, data: Data | UUID | str) -> UUID:
```

Comment:
The add_data function was settiing the association to the first data association if association was None. 
It sounds this mecanic disapear

Reply:
No still there, under validate association